infrastructure/k8s/readme.md

```diff
@@ -0,0 +1,255 @@
+```
+brew install helm
+brew install kubectx
+```
+
+Add stable official helm charts and repository.
+```
+helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+helm repo update
+```
+
+## Ingress
+
+Create namespace and change into the namespace.
+```
+kubectl create namespace ingress
+kubens ingress   
+```
+
+Install nginx ingress using helm chart.
+```
+helm upgrade --install ingress stable/nginx-ingress --namespace ingress
+```
+
+Wait for the public IP to be assigned.
+```
+watch --interval=1 kubectl --namespace ingress get services -o wide ingress-nginx-ingress-controller
+```
+
+Switch back to default namespace.
+```
+kubens default
+```
+
+# Certificate manager
+
+Install the CustomResourceDefinition resources separately
+```
+kubectl apply --validate=false -f https://raw.githubusercontent.com/jetstack/cert-manager/release-0.11/deploy/manifests/00-crds.yaml
+```
+
+Create the namespace for cert-manager
+```
+kubectl create namespace cert-manager
+kubens cert-manager
+```
+
+Add the Jetstack Helm repository
+```
+helm repo add jetstack https://charts.jetstack.io
+```
+
+Update your local Helm chart repository cache
+```
+helm repo update
+```
+
+Install the cert-manager Helm chart to enable support for lets-encrypt.
+```
+helm install cert-manager --namespace cert-manager --version v0.11.0 jetstack/cert-manager
+```
+
+## Monitoring
+To use Azure insights we need an analytics workspace to send data into. 
+```
+az resource create --resource-type Microsoft.OperationalInsights/workspaces \
+ --name coverServiceWorkspace \
+ --resource-group ${res} \
+ --location ${region} \
+ --properties '{}' -o table
+```
+ 
+First get the resource id of the workspace you created, by running
+
+```
+workspaceresoid=$(az resource show --resource-type Microsoft.OperationalInsights/workspaces --resource-group ${res} --name coverServiceWorkspace --query "id" -o tsv)
+```
+
+Next enable the monitoring add-on by running the command below.
+```
+az aks enable-addons --resource-group ${res} \
+    --name ${ksname} \
+    --addons monitoring \
+    --workspace-resource-id ${workspaceresoid}
+```
+
+
+Send pod container logs to insights.
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+    name: containerHealth-log-reader
+rules:
+    - apiGroups: ["", "metrics.k8s.io", "extensions", "apps"]
+      resources:
+         - "pods/log"
+         - "events"
+         - "nodes"
+         - "pods"
+         - "deployments"
+         - "replicasets"
+      verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+    name: containerHealth-read-logs-global
+roleRef:
+    kind: ClusterRole
+    name: containerHealth-log-reader
+    apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: User
+  name: clusterUser
+  apiGroup: rbac.authorization.k8s.io
+```
+
+```
+kubectl apply -f logreader-rbac.yaml
+```
+
+## Horizontal pod autoscaler (HPA)
+
+@TODO: description missing
```

Comment:
Missing

Reply:
Yeah... not done that part in to AKS yet... so more writing will come here.